docs/docker/docker-images.md

```diff
@@ -0,0 +1,218 @@
+
+### Build image using docker file
+The `docker build -t ng2-book-finder .` command takes the `Dockerfile` in the current directory, and builds an image called `ng2-book-finder` on your local machine.
+
+First Docker checks to make sure it has everything it needs to build.
+
+```
+ng2-book-finder (feature/docker-image) $ docker build -t ng2-book-finder .
+Sending build context to Docker daemon   214 kB
+```
+
+Then, Docker loads with the `node:4.2-slim` image, as the image is not present on the current docker-machine, it downloads the image.
+
+```
+Step 1 : FROM node:4.2-slim
+4.2-slim: Pulling from library/node
+Status: Downloaded newer image for node:4.2-slim
+ ---> 264411d7f3f9
+```
+
+The below steps are performed for building the image.
+ - Expose the port
+ - set the required working directory
+ - copy required files for the image.
+ - Do `npm install`.
+ - And then compile the typescript to javascript.
+
+```
+Step 2 : EXPOSE 3000
+ ---> Running in 71914456ebf1
+ ---> bff560e19942
+Removing intermediate container 71914456ebf1
+Step 3 : WORKDIR /dist
+ ---> Running in 5405a8603861
+ ---> b5f28bfb0b84
+Removing intermediate container 5405a8603861
+Step 4 : COPY ./package.json /dist
+ ---> 1ff76e6a087e
+Removing intermediate container c0af467292f6
+Step 5 : COPY ./app /dist/app
+ ---> c03cb65d13f7
+Removing intermediate container 765411be2826
+Step 6 : COPY ./assets /dist/assets
+ ---> 675d1ba05841
+Removing intermediate container 08bc09448d01
+Step 7 : COPY ./index.html /dist
+ ---> e82d5f412d57
+Removing intermediate container 725a2e08b91d
+Step 8 : COPY ./favicon.ico /dist
+ ---> 1789b98509cd
+Removing intermediate container e71d39b771e0
+Step 9 : RUN npm install
+ ---> Running in 31392f0e6e12
+ ---> 3849cb8f39aa
+Removing intermediate container 31392f0e6e12
+Step 10 : RUN npm run tsc:w
+ ---> Running in def4e79b2e9f
+npm info it worked if it ends with ok
+npm info using npm@2.14.12
+npm info using node@v4.2.6
+npm info pretsc:w ng2-book-finder@1.0.0
+npm info tsc:w ng2-book-finder@1.0.0
+
+> ng2-book-finder@1.0.0 tsc:w /dist
+> tsc -w
+Removing intermediate container def4e79b2e9f
+Step 11 : CMD npm run lite
+ ---> Running in 29a5657a4a25
+ ---> 3e957e2c6985
+Removing intermediate container 29a5657a4a25
+Successfully built 3e957e2c6985
+```
+
+Type `docker images` and press RETURN.
+This command, lists the images you have locally.
+
+```
+ng2-book-finder (feature/docker-image) $ docker images
+REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
+ng2-book-finder     latest              3e957e2c6985        30 seconds ago      356.1 MB
+node                4.2-slim            264411d7f3f9        5 weeks ago         205.4 MB
+```
+
+## Managing the image in Docker Registry
+
+In `docker-hub` we have created an organization called as `introductiontodocker` and within that organization we are having a repository `ng2-book-finder`.
```

Comment:
Can we have link to docker-hub  organization repository, so it will be easy to navigate and verify